src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -260,12 +253,40 @@ bool Terminal::SendKeyEvent(const WORD vkey, const ControlKeyStates states)
 
     const bool manuallyHandled = ch != UNICODE_NULL;
 
-    KeyEvent keyEv{ true, 0, vkey, 0, ch, states.Value() };
+    KeyEvent keyEv{ true, 0, vkey, scanCode, ch, states.Value() };
     const bool translated = _terminalInput->HandleKey(&keyEv);
 
     return translated && manuallyHandled;
 }
 
+WORD Terminal::_ScanCodeFromVirtualKey(const WORD vkey) noexcept
+{
+    return LOWORD(MapVirtualKeyW(vkey, MAPVK_VK_TO_VSC));
+}
+
+wchar_t Terminal::_CharacterFromKeyEvent(const WORD vkey, const WORD scanCode, const ControlKeyStates states) noexcept
+{
+    const auto sc = scanCode != 0 ? scanCode : _ScanCodeFromVirtualKey(vkey);
+
+    // We might want to use GetKeyboardState() instead of building our own keyState.
+    // The question is whether that's necessary though. For now it seems to work fine as it is.
+    BYTE keyState[256] = {};
+    keyState[VK_SHIFT] = states.IsShiftPressed() ? 0x80 : 0;
+    keyState[VK_CONTROL] = states.IsCtrlPressed() ? 0x80 : 0;
+    keyState[VK_MENU] = states.IsAltPressed() ? 0x80 : 0;
+
+    // Technically ToUnicodeEx() can produce arbitrarily long sequences of diacritics etc.
+    // Since we only handle the case of a single UTF-16 code point, we can set the buffer size to 2 though.
+    constexpr size_t bufferSize = 2;
+    wchar_t buffer[bufferSize];
+
+    // wFlags: If bit 2 is set, keyboard state is not changed (Windows 10, version 1607 and newer)
+    const auto result = ToUnicodeEx(vkey, sc, keyState, buffer, bufferSize, 0b100, nullptr);
+
+    // TODO: We're only handling basic UTF-16 code points right now, since that's the only thing KeyEvent supports.

```

Comment:
Could you create a GitHub issue to track this? Then change the beggining of this comment to "TODO GH#<issue>"